docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -0,0 +1,435 @@
+---
+title: "graphql"
+sidebar_position: 7
+description:
+  "This documentation cover methods related to GraphQL exposed by
+  `@front-commerce/core`."
+---
+
+<p>{frontMatter.description}</p>
+
+## `createGraphQLModule`
+
+Create a GraphQL module and inject it into Front-Commerce's GraphQL server
+
+```ts
+import { createGraphQLModule } from "@front-commerce/core/graphql";
+
+export default createGraphQLModule({
+  namespace,
+  modules,
+});
+```
+
+Arguments:
+
+- `namespace` (`string`): The namespace of the GraphQL module
+- `modules` (`modules[]`): The modules to inject in namespace
+
+_Example_
+
+```ts
+import { createGraphQLModule } from "@front-commerce/core/graphql";
+import AcmeModule from "./acmeModule";
+
+export default createGraphQLModule({
+  namespace: "Acme",
+  modules: [AcmeModule],
+});
+```
+
+## `readGraphQLModules`
+
+Read and parse GraphQLModules generated by codegen.
+
+_Example_
+
+```ts
+import { readGraphQLModules } from "@front-commerce/core/graphql";
+
+const generatedSchemaModules = await readGraphQLModules();
+```
+
+## `schema`
+
+Reads the generated GraphQL schema
+
+```ts
+import { schema } from "@front-commerce/core/graphql";
+
+const acmeSchema = await schema();
+```
+
+## `createEnvelop`
+
+:::caution
+
+This is an internal API to inject context into external modules. It may needs
+some modification to fetch your usecase
+
+See
+[`@front-commerce/remix/playground/yoga.ts`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/remix/playground/yoga.ts?ref_type=heads)
+for usage example.
+
+:::
+
+## `createGraphQLApi`
+
+Create a GraphQL method to execute GraphQL queries and mutations
+
+```ts
+const api = createGraphQLApi(request, user, config);
+```
+
+Arguments:
+
+- `request` (`Request`): The request object
+- `user` (`User`): The user context
+- `config` (`Config`): The application configuration
+
+_Example_
+
+```ts
+import { createGraphQLApi } from "@front-commerce/core/graphql";
+import { AcmeQuery } from "~/graphql/graphql";
+
+const AcmeGraphQLApi = createGraphQLApi(request, user, config);
+
+const response = await AcmeGraphQLApi(AcmeQuery);
+```
```

Comment:
nit: This example seems redundant with the other "example" just above.

suggestion: Remove the first one put the more complete example in its place instead.